src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using Microsoft.VisualStudio.Shell;
+using Recommender = Microsoft.DataAI.NuGetRecommender.Contracts;
+using System.Diagnostics;
+using NuGet.PackageManagement.Telemetry;
+using NuGet.Common;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Represents a package feed which recommends packages based on currently loaded project info
+    /// </summary>
+    public class RecommenderPackageFeed : IPackageFeed
+    {
+        public bool IsMultiSource => false;
+
+        private readonly SourceRepository _sourceRepository;
+        private readonly IEnumerable<PackageCollectionItem> _installedPackages;
+        private readonly IEnumerable<PackageCollectionItem> _dependentPackages;
+        private readonly IEnumerable<string> _targetFrameworks;
+        private readonly IPackageMetadataProvider _metadataProvider;
+        private readonly Common.ILogger _logger;
+        private readonly INuGetTelemetryService _telemetryService;
+
+        Recommender.IVsNuGetPackageRecommender NuGetRecommender { get; set; }
+
+        public RecommenderPackageFeed(
+            SourceRepository sourceRepository,
+            IEnumerable<PackageCollectionItem> installedPackages,
```

Comment:
The packages can change during the life time of a package manager UI. 
So that'd imply that the recommendations should change as well.

Reply:
That was my intent when I added the check for recommendations in the Refresh method. Are you asking for a different change here? Can you clarify?

Reply:
Does this collection get updated when the installed packages change?

Reply:
I think what happens is the Search logic creates a new `RecommenderPackageFeed`. This is relying on the fact that Installing a package triggers a refresh of the PMUI, which triggers the aforementioned Search logic.

Reply:
I looked at the different ways RecommendPackagesAsync could be called, and didn't find any path where the installed packages could have changed without creating a new RecommenderPackageFeed.

Reply:
Sounds good. Resolving.